lib/models/organisation.js

```diff
@@ -217,98 +217,108 @@ schema.post('init', function handleInit() {
 schema.pre('save', async function preSave(next) {
   const User = getConnection().model('User');
   const Role = getConnection().model('Role');
+  const organisation = this;
+  const objectId = mongoose.Types.ObjectId;
 
   // https://github.com/Automattic/mongoose/issues/1474
   this.wasNew = this.isNew;
 
-  const organisation = this;
-  const objectId = mongoose.Types.ObjectId;
-
   if (!isUndefined(this._oldExpiration) && this.expiration !== this._oldExpiration) {
     this.expirationNotifications.expirationNotificationSent = EMAIL_NOOP;
     this.expirationNotifications.weekBeforeNotificationSent = EMAIL_NOOP;
   }
 
+  if (!isUndefined(this._oldTimezone) && this.timezone !== this._oldTimezone) {
+    await update$dteTimezoneInDB(organisation._id, this.timezone);
+  }
+
+  await User.updateMany(
+    { ownerOrganisation: organisation._id },
+    { ownerOrganisationSettings: this.settings },
+    { runValidators: false },
+  );
+
   // https://github.com/Automattic/mongoose/issues/1474
-  if (organisation.wasNew) {
-    const createAdminRole = await Role.create({
-      title: 'Admin',
-      owner_id: organisation.owner,
-      organisation: organisation._id,
-      scopes: [ALL]
-    });
+  if (!organisation.wasNew) {
+    next();
+    return;
+  }
 
-    const createObserverRole = await Role.create({
-      title: 'Observer',
-      owner_id: organisation.owner,
-      organisation: organisation._id,
-      scopes: [
-        VIEW_PUBLIC_DASHBOARDS,
-        VIEW_PUBLIC_VISUALISATIONS,
-        VIEW_PUBLIC_JOURNEYS
-      ]
-    });
+  const createAdminRole = await Role.create({
```

Comment:
Why are we creating roles before finding the user? What if we don't find the owner by organisation.owner.

Reply:
We want these two roles to exist regardless of whether we can find the owner. This prevent issues caused by concurrency where the owner is deleted while we're creating the organisation.